includes/http/WordPress_HTTP_Client.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * WordPress AI Client HTTP Client Adapter
+ *
+ * @package WordPress\AI_Client
+ * @since n.e.x.t
+ */
+
+namespace WordPress\AI_Client\HTTP;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use WordPress\AiClient\Providers\Http\Contracts\ClientWithOptionsInterface;
+use WordPress\AiClient\Providers\Http\DTO\RequestOptions;
+use WordPress\AiClient\Providers\Http\Exception\NetworkException;
+
+/**
+ * PSR-18 HTTP Client adapter using WordPress HTTP API
+ *
+ * This adapter allows WordPress HTTP functions to be used
+ * as a PSR-18 compliant HTTP client.
+ *
+ * @since n.e.x.t
+ */
+class WordPress_HTTP_Client implements ClientInterface, ClientWithOptionsInterface {
+
+	/**
+	 * Response factory instance.
+	 *
+	 * @var ResponseFactoryInterface
+	 */
+	private $response_factory;
+
+	/**
+	 * Stream factory instance.
+	 *
+	 * @var StreamFactoryInterface
+	 */
+	private $stream_factory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param ResponseFactoryInterface $response_factory PSR-17 Response factory.
+	 * @param StreamFactoryInterface   $stream_factory   PSR-17 Stream factory.
+	 */
+	public function __construct( ResponseFactoryInterface $response_factory, StreamFactoryInterface $stream_factory ) {
+		$this->response_factory = $response_factory;
+		$this->stream_factory   = $stream_factory;
+	}
+
+	/**
+	 * Sends a PSR-7 request and returns a PSR-7 response.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return ResponseInterface The PSR-7 response.
+	 *
+	 * @throws NetworkException If the WordPress HTTP request fails.
+	 */
+	public function sendRequest( RequestInterface $request ): ResponseInterface {
+		$args = $this->prepare_wp_args( $request );
+		$url  = (string) $request->getUri();
+
+		/** Ignoring PHPStan for WordPress-specific array structure. @phpstan-ignore-next-line */
+		$response = \wp_remote_request( $url, $args );
+
+		if ( \is_wp_error( $response ) ) {
+			$message = sprintf(
+				'Network error occurred while sending request to %s: %s',
+				$url,
+				$response->get_error_message()
+			);
```

Comment:
Minor suggestion: Include the request method here as well.
```suggestion
			$message = sprintf(
				'Network error occurred while sending %s request to %s: %s',
				$request->getMethod(),
				$url,
				$response->get_error_message()
			);
```

Reply:
Resolved in 7943ed6